orca-web/src/main/groovy/com/netflix/spinnaker/orca/controllers/OperationsController.groovy

```diff
@@ -251,6 +268,16 @@ class OperationsController {
     [ref: "/pipelines/${pipeline.id}".toString()]
   }
 
+  private Map<String, String> markPipelineFailed(Map config, Throwable t) {
+    injectPipelineOrigin(config)
+    def json = objectMapper.writeValueAsString(config)
+    log.info('requested pipeline: {}', json)
+
+    def pipeline = executionLauncher.logFailedExecution(PIPELINE, json, t)
```

Comment:
This reads slightly off ... `logFailedExecution` but it returns a new execution? This would imply that the method is doing something other than logging.

Reply:
Perhaps easiest addressed by `s/logFailedExecution()/fail()` or something along those lines?

Reply:
Maybe this is something I can clarify by changing the name of `logFailedExecution`?  We are creating an execution (so that the error surfaces in the execution history), but rather than trying to start it we're just putting it in `TERMINAL` status.  Would it be more clear to call this `createFailedExecution` or something similar?

Reply:
Yup, I'll change the name to `fail()`, I agree that's clearer